### ISSUE TITLE
setState not refresh StaggeredGridView

### ISSUE THREAD

**Issue by littleMeng** [letsar/flutter_staggered_grid_view#21]
I want change the Container color when click the item. StaggeredGridView.count not work, but GridView working well. Are they different?

```
class DeviceTypeItem extends StatefulWidget {
  final List<DeviceTypeList> deviceTypes;
  final Function onSelected;

  DeviceTypeItem(this.deviceTypes, this.onSelected);

  @override
  _DeviceTypeItemState createState() => _DeviceTypeItemState();
}

class _DeviceTypeItemState extends State<DeviceTypeItem> {
  int _selectIndex = -1;

  _DeviceTypeItemState();

  _selectDevice(int index) {
    setState(() {
      _selectIndex = index;
    });
//    onSelected(deviceTypes[_selectIndex].deviceTypeName);
  }

  _tmpData() {
    return List.generate(10, (int index) => GestureDetector(
            child: Container(
              child: Text(widget.deviceTypes[0].deviceTypeName),
              color: Color(_selectIndex == index ? ColorHelp.primaryValue : ColorHelp.white),
              padding: EdgeInsets.only(top: 4, bottom: 4),
              alignment: Alignment.center,
            ),
            onTap: ()=> _selectDevice(index),
          ));
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(12),
//      child: GridView.count(
//        crossAxisCount: 3,
//        children: _tmpData(),
//        shrinkWrap: true,
//      ),

      child: StaggeredGridView.count(
        crossAxisCount: 3,
        children: _tmpData(),
        staggeredTiles: List.generate(10, (int index) => StaggeredTile.fit(1)),
        shrinkWrap: true,
      ),
    );
  }
}
```

**Comment by Himhotep** [comment]
I have no tried this example code but I think the same.

File: https://github.com/letsar/flutter_staggered_grid_view/blob/master/example/lib/example_6.dart
onLongPress changes _selectedIndex within a call to setState but nothing happens.

**Comment by developereranga** [comment]
@letsar yes when i call setState after changing the data StaggeredGridView not updating..

**Comment by ajkerfaisal** [comment]
facing the same issue. it adds or removes items but refreshing existing cells not happening.

**Comment by letsar** [comment]
Hi, I think I have also fixed this for the #23 .
Can you check with the master branch if this is still happening?

**Comment by littleMeng** [comment]
@letsar  I ensure use the newest master version, but It doesn't work.

**Comment by letsar** [comment]
@littleMeng can you provide a standalone example please so that I can test it?
I tested the master branch with setState modifications and existing cells are refreshing.

**Comment by Himhotep** [comment]
example_6.dart works fine now.
thanks!!!

**Comment by letsar** [comment]
Published in version 0.2.6.
